webapp/src/components/modals/create_event_form.tsx

```diff
@@ -72,13 +97,12 @@ export default function CreateEventForm(props: Props) {
         // add required field validation
 
         setSubmitting(true);
-        createEvent(formValues).then(({error}) => {
-            if (error) {
-                handleError(error);
-                return;
-            }
-
+        createEvent(formValues).then((_data) => {
             handleClose();
+        }).catch((response) => {
+            if (response.status_code >= 400) {
+                handleError(response.message);
+            }
         });
```

Comment:
Having the `response` as the thrown value "caught" here seems kind of counter-intuitive. I would expect an `Error` (which is essentially `{message: 'some error'`}) to be the caught value. JS exceptions really suck, and typescript essentially has no help for them